### ISSUE TITLE
Compose not standing up web container

### ISSUE THREAD

**Issue by peter-clemenko** [lanmaster53/recon-ng#125]
The web container is not standing up when running docker compose.

**Comment by lanmaster53** [comment]
Interesting. This should all happen automatically. It does for testing and I've never had anyone else report an issue like this. I believe it has something to do with this line in your error: `': No such file or directory can't execute 'python3`. What's all that about? It needs python3 for anything to work.

**Comment by peter-clemenko** [comment]
Shouldn't it build python in to the container in the dockerfile?

**Comment by lanmaster53** [comment]
Yes, which is why none of this makes sense to me. I would need to see every step you're taking an understand more about your environment to be able to help any further. That error alone tells me something is wrong, but I have no context beyond that.

**Comment by peter-clemenko** [comment]
I'm using windows 10. The build itself seems to be dependent on an existing reconng install in linux as indicated by having to create the .recon-ng directory. Try building it from scratch from a clean windows 10

**Comment by lanmaster53** [comment]
Ah. That explains it. Recon-ng does not support windows. I see that I don't have that in the Wiki anywhere. I should probably add that. I apologize for the inconvenience. On a side note, there is a pull request to add windows support (https://github.com/lanmaster53/recon-ng/pull/104). I don't know that I'll ever merge it as I am still considering whether or not I want to go that route, but you can try that is you like. I recommend using Docker. It's the easiest and most full featured.

**Comment by peter-clemenko** [comment]
That may be the case, but the problem is while trying to build and run the docker container. I'm encountering these problems after having to hack around building and running the docker container.

**Comment by lanmaster53** [comment]
Not only does Recon-ng not run on windows directly, but the Dockerfile and docker-compose files both use paths that aren't valid in windows. Perhaps that's the issue?

**Comment by peter-clemenko** [comment]
Quite possibly, but if that's the case, it needs to be fixed.

**Comment by lanmaster53** [comment]
What needs to be fixed? We don't support Windows. I intend to update the documentation if that's what you mean.

**Comment by peter-clemenko** [comment]
The ability to stand up/build the docker containers on a windows host? The docker containers shouldn't be affected by the host being windows.

**Comment by lanmaster53** [comment]
It would require separate Dockerfile / docker-compose files for linux and windows because the mapped paths would be different, etc. I personally don't use windows, so I have no need to create them for that environment, but if you want to put them together and submit a PR, I'd be happy to provide them to others.

**Comment by peter-clemenko** [comment]
Can you just set it up so that it auto builds and pushes to a container registry and the compose has the option to pull from the container registry?

**Comment by lanmaster53** [comment]
I suppose it's possible, but I don't have the cycles to learn and do that right now. I don't know much about the Docker ecosystem outside of using Docker Desktop to build and run containers.

**Comment by peter-clemenko** [comment]
Literally all you have to do is set up an account on docker hub, add your github repo, and add the dockerfile, and it will build on each push automatically.
